src/resmom/mom_comm.c

```diff
@@ -6156,14 +6525,459 @@ tm_request(int fd, int version)
+						free(temptaskid);
+						free(tempnid);
+						free(streamlist);
+						tpp_mcast_close(mtfd);
+						goto done;
+					}
+				}
+
+				ret = diswsi(mtfd, myvnodeid);
+				if (ret != DIS_SUCCESS) {
+					arrayfree(argv);
+					arrayfree(envp);
+					free(node_list);
+					free(tempinfo);
+					free(temptaskid);
+					free(tempnid);
+					free(streamlist);
+					tpp_mcast_close(mtfd);
+					goto done;
+				}
+				if (found_empty_string) {
+					ret = diswui(mtfd, argc);
+					if (ret != DIS_SUCCESS) {
+						arrayfree(argv);
+						arrayfree(envp);
+						free(node_list);
+						free(tempinfo);
+						free(temptaskid);
+						free(tempnid);
+						free(streamlist);
+						tpp_mcast_close(mtfd);
+						goto done;
+					}
+					for (i = 0; i < argc; i++) {
+						ret = diswst(mtfd, argv[i]);
+						if (ret != DIS_SUCCESS) {
+							arrayfree(argv);
+							arrayfree(envp);
+							free(node_list);
+							free(tempinfo);
+							free(temptaskid);
+							free(tempnid);
+							free(streamlist);
+							tpp_mcast_close(mtfd);
+							goto done;
+						}
+					}
+				} else {
+				  	for (i = 0; argv[i]; i++) {
+						ret = diswst(mtfd, argv[i]);
+						if (ret != DIS_SUCCESS) {
+							arrayfree(argv);
+							arrayfree(envp);
+							free(node_list);
+							free(tempinfo);
+							free(temptaskid);
+							free(tempnid);
+							free(streamlist);
+							tpp_mcast_close(mtfd);
+							goto done;
+						}
+					}
+					ret = diswst(mtfd, "");
+					if (ret != DIS_SUCCESS) {
+						arrayfree(argv);
+						arrayfree(envp);
+						free(node_list);
+						free(tempinfo);
+						free(temptaskid);
+						free(tempnid);
+						free(streamlist);
+						tpp_mcast_close(mtfd);
+						goto done;
+					}
+				}
+				for (i = 0; envp[i]; i++) {
+					ret = diswst(mtfd, envp[i]);
+					if (ret != DIS_SUCCESS) {
+						arrayfree(argv);
+						arrayfree(envp);
+						free(node_list);
+						free(tempinfo);
+						free(temptaskid);
+						free(tempnid);
+						free(streamlist);
+						tpp_mcast_close(mtfd);
+						goto done;
+					}
+				}
+				ret = (dis_flush(mtfd) == -1) ?
+					DIS_NOCOMMIT : DIS_SUCCESS;
+				if (ret != DIS_SUCCESS) {
+					arrayfree(argv);
+					arrayfree(envp);
+					free(node_list);
+					free(tempinfo);
+					free(temptaskid);
+					free(tempnid);
+					free(streamlist);
+					tpp_mcast_close(mtfd);
+					goto done;
+				}
+			}
+
+			/*
+			** If there is no need to send to sisters
+			** we can reply now, we're done
+			*/
+			if (momcnt == 0) {
+				(void)tm_reply(fd, version, i, event);
+				/* sending the number of task ids */
+				(void)diswui(fd, index);
+
+				/* send the task ID and the node it is on */
+				for (k = 0; k < index; k++) {
+					(void)diswui(fd, temptaskid[k]);
+					(void)diswui(fd, tempnid[k]);
+				}
+				(void)dis_flush(fd);
+				/* Reset the task counts */
+				tempinfo->ji_num_sisters = 0;
+				tempinfo->ji_taskid_index = 0;
+				/* These aren't needed anymore */
+				free(temptaskid);
+				free(tempnid);
```

Comment:
should you free tempinfo here?

Reply:
No, tempinfo is freed when the job structure is freed

Reply:
Now I see what you are asking, yes, I can also free tempinfo here.